### PR TITLE
Fix build problems in NixOs

### PR DESCRIPTION
**Pull request by ericbmerritt**

### CODE REVIEW

File: lib/cmd/cmd_common.ml [hunk 1]
```diff
@@ -3,6 +3,8 @@ open Core_extended.Std
 open Async.Std
 open Deferred.Monad_infix
 
+exception Execution_failed of Error.t
```
Comment (samuelrivas) [inline lib/cmd/cmd_common.ml]: Still using exceptions?

File: lib/cmd/cmd_common.ml [hunk 2]
```diff
@@ -31,23 +33,61 @@ let result_guard def_fn =
...
+  | None -> Sys.getcwd ()
+
+let sh
+  : ?working_dir:String.t -> ?shell:String.t List.t -> String.t ->
+    (String.t, Exn.t) Deferred.Result.t =
+  fun ?working_dir ?(shell = ["bash"; "-c"]) cmd ->
+    get_cwd working_dir
+    >>= fun cwd ->
+    let actual_args = List.concat [shell; [cmd]] in
+    Process.run ~working_dir:cwd ~prog:"/usr/bin/env" ~args:actual_args ()
```
Comment (samuelrivas) [inline lib/cmd/cmd_common.ml]: Why do we need to run the program using env?

File: lib/cmd/cmd_common.ml [hunk 3]
```diff
@@ -31,23 +33,61 @@ let result_guard def_fn =
...
     Command.async_basic ~summary:desc
       Command.Spec.empty
-      (cmd_monitor ~exn format) in
-  (name, command)
-
-
+      (fun () -> 
+         result_guard 
+           (fun () -> simply_print_response ?working_dir ?shell cmd )) in
+  (name, command)
```
Comment (samuelrivas) [inline lib/cmd/cmd_common.ml]: Style. I prefer to have line jumps, but obviously is up for discussion

Comment (ericbmerritt) [inline lib/cmd/cmd_common.ml]: Line jumps? (this should have a newline at the end, absolutely. Also, I am not a super fan of the nested anonymous functions here.

File: lib/cmd/cmd_common.ml (hunk 1 above)
Comment (ericbmerritt) [inline lib/cmd/cmd_common.ml]: Its still used pretty ubiquitously in our stuff. Until we go through and change it all, I will keep using it. I went through the exercise of trying to migrate to polymorphic variants again a month or so ago and ran into problems. I am really open to the idea of you going through and  converting everything  :) I have failed twice now.

File: lib/cmd/cmd_common.ml (hunk 2 above)
Comment (ericbmerritt) [inline lib/cmd/cmd_common.ml]: Because we can't assume hard paths. At least not with nixos in play.

Comment (samuelrivas) [inline lib/cmd/cmd_common.ml]: But you can assume it accessible in the path. This is how I do it for nixos: https://github.com/samuelrivas/assorted-scripts/blob/master/nix/post-fixup.sh#L28

That means that the script/program would work normally (i.e. searching through the PATH) when run as is, but will use the exact dependency if installed through nixos

Reference:
https://nixos.org/wiki/Nix_Runtime_Environment_Wrapper

File: lib/cmd/cmd_common.ml (hunk 1 above)
Comment (samuelrivas) [inline lib/cmd/cmd_common.ml]: Fair enough

**Comment by ericbmerritt** [comment]
Updated @samuelrivas

File: lib/trv.ml [hunk 4]
```diff
@@ -1,16 +1,6 @@
...
-module Build = Build
-
-let command =
-  Command.group ~summary:"Base tooling system for Afiniate projects"
-  [Build.desc;
-   Opam.desc]
-
-let () =
-  Command.run ~version:"1.0" ~build_info:"" command
+module Build = Build
```
Comment (samuelrivas) [inline lib/trv.ml]: line jump.

**Comment by samuelrivas** [comment]
Looks good other than that minor comment in trv.ml

**Comment by ericbmerritt** [comment]
@samuelrivas fixed

**Pull request merged**